src/Tasks/Microsoft.NET.Build.Extensions.Tasks/msbuildExtensions/Microsoft/Microsoft.NET.Build.Extensions/Microsoft.NET.Build.Extensions.NETFramework.targets

```diff
@@ -57,8 +67,20 @@ Copyright (c) .NET Foundation. All rights reserved.
     <NETBuildExtensionsError Condition="'$(DependsOnNETStandard)' == 'true' AND '$(NETStandardInbox)' != 'true' AND '$(_UsingOldSDK)' == 'true'"
                              ResourceName="UnsupportedSDKVersionForNetStandard20"/>
     
+    <!--
+      .NET 4.7.1 has support for .NET Standard 2.0 built-in, so most of the facades aren't necessary.  However, the assembly versions of System.Net.Http and
+      System.IO.Compression in .NET 4.7.1 are still 4.0.0.0.  This means that the versions from the contract NuGet packages (which are versioned 4.1.0.0 or
+      higher) would be preferred to the in-box version (which is newer).  So if there is a dependency on the "contract" version of these DLLs, or on
+      netstandard.dll, we use the 4.2.0.0 version of the DLLs from the .NET Standard 2.0 "facades".  (Though these DLLs are not actually facades, they
+      contain the implementation.)
+      -->
+    <ItemGroup Condition="'$(_TargetFrameworkVersionWithoutV)' == '4.7.1'">
+      <_NETStandardLibraryNETFrameworkLib Include="$(MSBuildThisFileDirectory)\net461\lib\System.Net.Http.dll" Condition="'$(DependsOnNuGetHttp)' == 'true' Or '$(DependsOnNETStandard)' == 'true'"/>
```

Comment:
Is it worth the extra complexity to try and detect references to these? Why not just always include them in 4.7.1 like we do for the earlier versions?

Reply:
+1. I don't like that we're adding all this code to workaround a framework bug.

Reply:
What is the case where DependsOnNETStandard==false (which is actually system.runtime >= some-ver or netstandard) but DependsOnNugetXxx==true. Which test covers that?

Reply:
@weshaggard @nguerrera 

> Is it worth the extra complexity to try and detect references to these? Why not just always include them in 4.7.1 like we do for the earlier versions?

Should we include these libraries in .NET 4.7.1 apps that don't reference .NET Standard assemblies at all?  That's what this prevents.  We can't just use the existing logic to detect usage of .NET Standard >= 1.5, because we need the updated libraries if they are used by a .NET Standard 1.4 or lower library too.

Reply:
I think we should use whatever logic we are using to detect netstandard for both 2.0 and for earlier. 

@alexghi will give you more data but we believe this set might be larger now.